core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -254,10 +256,18 @@ private[spark] object SortShuffleManager extends Logging {
     val executorComponents = ShuffleDataIOUtils.loadShuffleDataIO(conf).executor()
     val extraConfigs = conf.getAllWithPrefix(ShuffleDataIOUtils.SHUFFLE_SPARK_CONF_PREFIX)
         .toMap
+    val pluginOutputTrackingEnabled = conf.get(config.SHUFFLE_IO_PLUGIN_OUTPUT_TRACKING_ENABLED)
+    val trackerWorker = SparkEnv.get.mapOutputTracker.asInstanceOf[MapOutputTrackerWorker]
+    val updater = if (pluginOutputTrackingEnabled) {
+      Some(trackerWorker)
+    } else {
+      Option.empty[ShuffleUpdater]
+    }
     executorComponents.initializeExecutor(
       conf.getAppId,
       SparkEnv.get.executorId,
-      extraConfigs.asJava)
+      extraConfigs.asJava,
+      updater.asJava)
```

Comment:
And when to use the updater is left to each ShuffleExecutorComponent to decide.. right? Here we just provide the framework to make the updater available..

Reply:
Yes that is the expected usage of this API.